apis/v1alpha2/nginxproxy_types.go

```diff
@@ -388,6 +389,35 @@ type KubernetesSpec struct {
 	Service *ServiceSpec `json:"service,omitempty"`
 }
 
+// Patch defines a patch to apply to a Kubernetes object.
+type Patch struct {
+	// Type is the type of patch. Defaults to StrategicMerge.
+	//
+	// +optional
+	// +kubebuilder:default:=StrategicMerge
+	Type *PatchType `json:"type,omitempty"`
+
+	// Value is the patch data as raw JSON.
+	// For StrategicMerge and Merge patches, this should be a JSON object.
+	// For JSONPatch patches, this should be a JSON array of patch operations.
+	//
+	// +kubebuilder:validation:XPreserveUnknownFields
+	Value *apiextv1.JSON `json:"value"`
```

Comment:
Should this field be marked as optional?

Reply:
Good catch, it's not required but it is actually best practice so I'll add it